src/python/pants/core/goals/test.py

```diff
@@ -104,31 +134,72 @@ def from_fallible_process_result(
         xml_results: Snapshot | None = None,
         extra_output: Snapshot | None = None,
     ) -> TestResult:
-        return cls(
+        return TestResult(
+            exit_code=process_result.exit_code,
+            stdout=process_result.stdout.decode(),
+            stdout_digest=process_result.stdout_digest,
+            stderr=process_result.stderr.decode(),
+            stderr_digest=process_result.stderr_digest,
+            addresses=(address,),
+            output_setting=output_setting,
+            result_metadata=process_result.metadata,
+            coverage_data=coverage_data,
+            xml_results=xml_results,
+            extra_output=extra_output,
+        )
+
+    @staticmethod
+    def from_batched_fallible_process_result(
+        process_result: FallibleProcessResult,
+        batch: TestRequest.Batch[_TestFieldSetT, Any],
+        output_setting: ShowOutput,
+        *,
+        coverage_data: CoverageData | None = None,
+        xml_results: Snapshot | None = None,
+        extra_output: Snapshot | None = None,
+    ) -> TestResult:
+        return TestResult(
             exit_code=process_result.exit_code,
             stdout=process_result.stdout.decode(),
             stdout_digest=process_result.stdout_digest,
             stderr=process_result.stderr.decode(),
             stderr_digest=process_result.stderr_digest,
-            address=address,
+            addresses=tuple(field_set.address for field_set in batch.elements),
             output_setting=output_setting,
             result_metadata=process_result.metadata,
             coverage_data=coverage_data,
             xml_results=xml_results,
             extra_output=extra_output,
+            partition_description=batch.partition_metadata.description
+            if batch.partition_metadata
+            else None,
         )
 
     @property
     def skipped(self) -> bool:
         return self.exit_code is None or self.result_metadata is None
 
+    @property
+    def description(self) -> str:
+        if len(self.addresses) == 1:
+            return self.addresses[0].spec
+        else:
+            return f"{self.addresses[0].spec} and {len(self.addresses)-1} other files"
```

Comment:
Can the description not be "gotten" from some user-provided description? Or is that coming later?

Reply:
The user-provided description is used in the `message` method below. This code is the common bit between logging & the final success-vs-failure printed display.